### ISSUE TITLE
Cube only support

### ISSUE THREAD

**Issue by andrewgryan** [MetOffice/forest#148]
FOREST currently assumes cubes are backed by NetCDF files. To be general and appeal to a wider audience it should be trivial to add full fledged `iris.cube.Cube` support

**Comment by andrewgryan** [comment]
`cube.coord('forecast_reference_time')` can be used instead of the current `netCDF4.Dataset` approach

**Comment by andrewgryan** [comment]
```
2019-10-02 14:55:44,548 Error running application handler <bokeh.application.handlers.directory.DirectoryHandler object at 0x2b5c7ffd6940>: [Errno -51] NetCDF: Unknown file format: b'umnsaa_mlppe071'
File "_netCDF4.pyx", line 1743, in netCDF4._netCDF4._ensure_nc_success:
 Traceback (most recent call last):
  File "/home/h03/frrn/miniconda3/envs/forest_cli/lib/python3.6/site-packages/bokeh/application/handlers/code_runner.py", line 179, in run
    exec(self._code, module.__dict__)
  File "/net/home/h03/frrn/software/github/forest/forest/main.py", line 718, in <module>
    main()
  File "/net/home/h03/frrn/software/github/forest/forest/main.py", line 126, in main
    locator = disk.Locator(args.files)
  File "/net/home/h03/frrn/software/github/forest/forest/disk.py", line 250, in __init__
    key = str(load_initial_time(path))
  File "/net/home/h03/frrn/software/github/forest/forest/disk.py", line 138, in __call__
    return self.netcdf4_strategy(path)
  File "/net/home/h03/frrn/software/github/forest/forest/disk.py", line 144, in netcdf4_strategy
    with netCDF4.Dataset(path) as dataset:
  File "netCDF4/_netCDF4.pyx", line 2123, in netCDF4._netCDF4.Dataset.__init__
  File "netCDF4/_netCDF4.pyx", line 1743, in netCDF4._netCDF4._ensure_nc_success
OSError: [Errno -51] NetCDF: Unknown file format: b'umnsaa_mlppe071'
```

**Comment by rhattersley** [comment]
It depends what you intended the scope of this issue to be, but for certain categories of data it is addressed by #180.
